src/main/java/org/jenkinsci/plugins/docker/commons/tools/DockerToolInstaller.java

```diff
@@ -0,0 +1,168 @@
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.ProxyConfiguration;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.remoting.Callable;
+import hudson.remoting.VirtualChannel;
+import hudson.tools.ToolInstallation;
+import hudson.tools.ToolInstaller;
+import hudson.tools.ToolInstallerDescriptor;
+import hudson.util.IOUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+
+
+/**
+ * Download and install Docker CLI binary, see https://docs.docker.com/installation/binaries/
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class DockerToolInstaller extends ToolInstaller {
+
+    private final String version;
+
+    @DataBoundConstructor
+    public DockerToolInstaller(String label, String version) {
+        super(label);
+        this.version = version;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    @Override
+    public FilePath performInstallation(ToolInstallation toolInstallation, Node node, TaskListener listener) throws IOException, InterruptedException {
+
+        String os = node.getChannel().call(new Callable<String, IOException>() {
+            public String call() throws IOException {
+                String os = System.getProperty("os.name").toLowerCase();
+                String arch = System.getProperty("os.arch").contains("64") ? "x86_64" : "i386";
+                if(os.contains("linux"))  return "Linux/"+arch;
+                if(os.contains("windows"))   return "Windows/"+arch;
+                if(os.contains("mac"))   return "Darwin/"+arch;
+                throw new IOException("Failed to determine OS architecture "+os+":"+arch);
+            }
+        });
+
+        final URL url = new URL("https://get.docker.com/builds/"+os+"/docker-"+version);
+        FilePath install = preferredLocation(tool, node);
+
+        // (simplified) copy/paste from FilePath as hudson.FilePath.installIfNecessaryFrom do assume the URL points to be a zip/tar archive
+
+        FilePath timestamp = install.child(".timestamp");
+        URLConnection con = null;
+        long sourceTimestamp;
+        try {
+            con = ProxyConfiguration.open(url);
+            if (timestamp.exists()) {
+                con.setIfModifiedSince(timestamp.lastModified());
+            }
+            con.connect();
+            if (con instanceof HttpURLConnection
+                    && ((HttpURLConnection)con).getResponseCode() == HttpURLConnection.HTTP_NOT_MODIFIED) {
+                return install;
+            }
+            sourceTimestamp = con.getLastModified();
+
+        } catch (IOException x) {
+            if (install.exists()) {
+                // Cannot connect now, so assume whatever was last unpacked is still OK.
+                if (listener != null) {
```

Comment:
this is the way it is implemented in FilePath#installIfNecessaryFrom 
as noticed, this is mostly a copy/paste due to a jenkins-core limitation